src/sql/workbench/services/contextualization/common/serverContextualizationService.ts

```diff
@@ -91,4 +104,36 @@ export class ServerContextualizationService extends Disposable implements IServe
 			});
 		}
 	}
+
+	public async onGenerateServerContextualizationComplete(handle: number, serverContextualizationCompleteParams: azdata.contextualization.GenerateServerContextualizationCompleteParams): Promise<void> {
+		if (serverContextualizationCompleteParams.context) {
+			const activeQueryEditorInput = this._editorService.activeEditorPane.input as QueryEditorInput;
```

Comment:
This event can take a while, so there's no guarantee that the current active editor both has its context back yet (we might have gotten context for another query first) and that it's still the currently active one.

Overall I'm still not sure why you need this complete message. Was there something wrong with how you had it before where making a connection for an editor would go and trigger the call to fetch the context for that connection and send it at that point?

Reply:
I think I should remove the complete event and change the generate context notification back to a request. In the request return the context when generated back to the client side, and have the background thread continue to store them.

I'm finding that the notification with complete callback is a clunky implementation.